tests/callbacks/test_image_visualizer.py

```diff
@@ -44,3 +47,60 @@ def test_image_visualizer(interval: str):
     assert isinstance(image_visualizer.interval, Time)
     assert num_train_tables == (num_train_steps - 1) // image_visualizer.interval.value + 1
     assert num_eval_tables == 1
+
+
+@pytest.mark.skipif(not _WANDB_INSTALLED, reason='Wandb is optional')
+def test_wandb_and_image_visualizer(tmp_path):
+    import wandb
+
+    image_interval = 2
+    image_visualizer = ImageVisualizer(interval=f'{image_interval}ba')
+    wandb_logger = WandBLogger(init_kwargs={'dir': tmp_path})
+
+    dataset_size = 40
+    batch_size = 4
+
+    trainer = Trainer(model=SimpleConvModel(),
+                      callbacks=image_visualizer,
+                      loggers=wandb_logger,
+                      train_dataloader=DataLoader(RandomImageDataset(size=dataset_size), batch_size),
+                      eval_dataloader=DataLoader(RandomImageDataset(size=dataset_size), batch_size),
+                      max_duration='1ep')
+
+    trainer.fit()
+
+    assert wandb.run is not None
+    wandb_run_dir = Path(wandb.run.dir)
+    run_file = wandb_run_dir.parent / Path(f'run-{wandb.run.id}.wandb')
+
+    # delete trainer to force WandBLogger to clean up in post_close
+    del trainer
+
+    # Note, it is not clear how to correctly load this file, so we are just loading it as text
+    # and searching the text for expected strings
+    with open(run_file, encoding='latin-1') as _wandb_file:
+        all_run_text = _wandb_file.read()
+
+    train_metrics_accuracy_count = all_run_text.count('metrics/train/Accuracy')
+    eval_metrics_accuracy_count = all_run_text.count('metrics/eval/Accuracy')
+    eval_metrics_cross_entropy_count = all_run_text.count('metrics/eval/CrossEntropy')
+    train_loss_count = all_run_text.count('loss/train/total')
+
+    expected_number_train_loss_count = (dataset_size / batch_size) + 1  # wandb includes it in the file one extra time
+    expected_number_train_metrics_count = (dataset_size /
+                                           batch_size) + 2  # wandb includes it in the file two extra times
```

Comment:
why the extra time(s)?

Reply:
I'm not 100% sure because the serialized file is a bit difficult to read, but it looks like some summary stuff gets logged at the end